src/token/mod.rs

```diff
@@ -348,7 +348,7 @@ fn partial_tokens_to_tokens(mut tokens: &[PartialToken]) -> EvalexprResult<Vec<T
                 },
                 PartialToken::Literal(literal) => {
                     cutoff = 1;
-                    if let Ok(number) = literal.parse::<IntType>() {
+                    if let Ok(number) = literal.parse_dec_or_hex::<IntType>() {
```

Comment:
Can we do this without `num-traits` by using `IntType::from_str_radix` instead?

By first checking if the prefix is "0x", and if yes, parsing as a hex, and if no, then continuing the if/else chain.

Reply:
yeah, probably better. I somehow tried to stay generic over the used IntType.